lib/TabTracker.js

```diff
@@ -45,6 +52,16 @@ TabTracker.prototype = {
     }
     tabs.removeListener("open", this.onOpen);
     Services.obs.removeObserver(this.onPerfLoadComplete, PERF_LOG_COMPLETE_NOTIF);
+    Services.obs.removeObserver(this.onCacheUpdate, CACHE_UPDATE_NOTIF);
```

Comment:
If we listen to the cache update notification, we would only obtain data if there is a cache update. This is incorrect: we want the history and bookmark sizes with or without cache updates.

A better way to obtain this data is to get ahold of the memoized query object instance and retrieving the data from there